### PR TITLE
Add instructions for developing with Toil Appliance (resolves #1611)

### LINKED ISSUE #1611: Improve documentation for developing with Toil Appliance
Currently instructions for how to build and upload your own docker images with the Toil appliance are vague/nonexistent. Knowing how to do this is vital for working on the AWS provisioner since one will need to build and test their own docker images in the cloud.

### PR DESCRIPTION
**Pull request by jessebrennan**
Adds step by step instructions for developing with the Toil Appliance.

### CODE REVIEW

**Comment by jessebrennan** [comment]
@cket review me plz

File: docs/contributing/contributing.rst [hunk 1]
```diff
@@ -128,9 +129,49 @@ as soon as a developer makes a commit or dirties the working copy they will no
...
+
+        $ make push_docker
+
+   which will upload the docker image to your Quay account. Take note of
+   the image's tag for the next step.
+
+6. Finally you will need to tell Toil from where to pull the Appliance
+   image you've created (it uses the Toil release you have installed by
+   default). To do this set the environment variable
+   ``TOIL_APPLIANCE_SELF`` to the url of your image. For more info see
```
Comment (cket) [inline docs/contributing/contributing.rst]: The envar is set to the tag of the image rather than the URL.

Comment (jessebrennan) [inline docs/contributing/contributing.rst]: Weird... I just checked the docs and you are of course right, but I had misread that and had been using the entire URL and everything was working fine.

Comment (cket) [inline docs/contributing/contributing.rst]: Interesting- navigating to `quay.io/ucsc_cgl/toil:3.5.0` redirects properly to `quay.io/repository/ucsc_cgl/toil?tag=3.5.0&tab=tags`. So it is a URL!

Comment (jessebrennan) [inline docs/contributing/contributing.rst]: So then is this ready to merge?

**Pull request merged**